### ISSUE TITLE
MemoryError when schedule

### ISSUE THREAD

**Issue by fyrestone** [agronholm/apscheduler#125]
Code Sample:
```
from __future__ import print_function
from apscheduler.schedulers.background import BackgroundScheduler


def scheduler():
    if not hasattr(scheduler, 'instance'):
        scheduler.instance = BackgroundScheduler()
        scheduler.instance.start()
    return scheduler.instance


s = scheduler()
s.add_job(lambda: print('hehe'), trigger='cron', second='*/2')
raw_input()
```
Insert a print in job.py:130
```
    def _get_run_times(self, now):
        """
        Computes the scheduled run times between ``next_run_time`` and ``now`` (inclusive).

        :type now: datetime.datetime
        :rtype: list[datetime.datetime]

        """
        run_times = []
        next_run_time = self.next_run_time
        while next_run_time and next_run_time <= now:
            run_times.append(next_run_time)
            print(len(run_times), next_run_time)
            next_run_time = self.trigger.get_next_fire_time(next_run_time, now)

        return run_times
```
Run sample code, sometimes(not every time) will print:
```
(826082, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826083, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826084, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826085, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826086, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826087, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826088, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826089, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826090, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826091, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826092, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
(826093, datetime.datetime(2016, 4, 8, 15, 25, 22, tzinfo=<DstTzInfo 'Asia/Shanghai' CST+8:00:00 STD>))
...
```

Maybe a solution:
```
    def _get_run_times(self, now):
        """
        Computes the scheduled run times between ``next_run_time`` and ``now`` (inclusive).

        :type now: datetime.datetime
        :rtype: list[datetime.datetime]

        """
        run_times = []
        next_run_time = self.next_run_time
        while next_run_time and next_run_time < now: # here replace <= with <, solve the bug
            run_times.append(next_run_time)
            next_run_time = self.trigger.get_next_fire_time(next_run_time, now)

        return run_times
```

**Comment by agronholm** [comment]
It seems like what you really want is IntervalTrigger with a 2 second interval. That said, are you saying CronTrigger is generating the same datetime over and over? Can you reproduce this reliably somehow?

**Comment by fyrestone** [comment]
This is a sample code, I use CronTrigger have problems so I reproduce the bug. apscheduler 3.1.0, Python 32bit 2.7.x in WIndows 10 & 7. BTW, I replaced <= with < in _get_run_times as a workaround.

**Comment by agronholm** [comment]
The script you provided works fine.

**Comment by fyrestone** [comment]
It fails some time but not every time.

**Comment by agronholm** [comment]
How often does that happen? I haven't found any way to make it fail.

**Comment by fyrestone** [comment]
My debug view (I tried many times to reproduce it), it shows that previous_fire_time == now and next_date never changed:
![image 1](https://cloud.githubusercontent.com/assets/6308809/14517374/353c77fc-023f-11e6-976d-a8d21cb66938.png)
![image 2](https://cloud.githubusercontent.com/assets/6308809/14517373/30f9cbfe-023f-11e6-8449-3d29113b5d29.png)

**Comment by agronholm** [comment]
Fixed in 21aa9a944247f4e05dfc91fc8dfed23b5d2b6d89.
